### PR TITLE
Implement methods lower()/islower() for unicode based on Cpython

### PR DESCRIPTION
**Pull request by densmirn**
The PR is continuation of [other one](https://github.com/numba/numba/pull/4263).

### CODE REVIEW

File: numba/unicode.py [hunk 1]
```diff
@@ -815,6 +818,30 @@ def unicode_strip_types_check(chars):
         raise TypingError('The arg must be a UnicodeType or None')
 
 
+@register_jitable
+def _handle_capital_sigma(s, l, idx):
+    #This algorithm is an approximate translation of:
```
Comment (Vyacheslav-Smirnov) [inline numba/unicode.py]: Suggest to specify what function do and returns here using """ """ doc-string

File: numba/unicode.py [hunk 2]
```diff
@@ -1148,6 +1175,126 @@ def impl(a):
...
+            if is_upper(code_point) or is_title(code_point):
+                return False
+            elif not cased and is_lower(code_point):
+                cased = True
+        return cased
+    return impl
+
+
+_always_false = register_jitable(lambda x:False)
+_ascii_is_lower = register_jitable(_is_lower(_Py_ISLOWER, _Py_ISUPPER, _always_false))
```
Comment (Vyacheslav-Smirnov) [inline numba/unicode.py]: As I understand, this is a method for unicode type - so input data can be unicode only?
Then if yes, no need to handle ascii...
Please correct me if it is wrong

File: numba/tests/test_unicode.py [hunk 3]
```diff
@@ -933,6 +941,39 @@ def pyfunc(x):
...
 
+    def test_islower(self):
+        pyfunc = islower_usecase
+        cfunc = njit(pyfunc)
+        lowers = [x.lower() for x in UNICODE_EXAMPLES]
+        extras = ["AA12A", "aa12a", "大AA12A", "大aa12a", "AAAǄA", "A 1 1 大"]
+
+        # Samples taken from CPython testing:
+        # https://github.com/python/cpython/blob/3.6/Lib/test/test_unicode.py#L571-L584
+        cpython = ['\u2167', '\u2177', '\U00010401', '\U00010427', '\U00010429',
```
[the remark this answers is not in the document]
Comment (densmirn) [reply numba/tests/test_unicode.py]: I agree to do that for own symbols, but the samples taken from CPython as is. So I think we don't need to convert these ones to human readable form.

File: numba/tests/test_unicode.py [hunk 4]
```diff
@@ -933,6 +941,39 @@ def pyfunc(x):
...
+                   '\U0001044E', '\U0001F40D', '\U0001F46F']
+        fourxcpy = [x * 4 for x in cpython]
+
+        for a in UNICODE_EXAMPLES + lowers + [""] + extras + cpython + fourxcpy:
```
[the remark this answers is not in the document]
Comment (densmirn) [reply numba/tests/test_unicode.py]: List UNICODE_EXAMPLES contains string with spaces and special characters.

File: numba/unicode.py (hunk 1 above)
Comment (densmirn) [reply numba/unicode.py]: I will do it.

File: numba/tests/test_unicode.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (densmirn) [reply numba/tests/test_unicode.py]: I will do it.

File: numba/unicode.py (hunk 2 above)
Comment (densmirn) [reply numba/unicode.py]: Unicode symbols include ASCII ones. So I agree to remove ASCII handling.

File: numba/tests/test_unicode.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (densmirn) [reply numba/tests/test_unicode.py]: I see only 2 places where the list is used. Moreover the list is duplicated in CPython testing: https://github.com/python/cpython/blob/3.6/Lib/test/test_unicode.py#L571-L600. So I think it doesn't make sense to move it to a single place to share between test cases.

**Comment by esc** [comment]
@densmirn thanks for this, I see some of the references to the cpython source code are referencing `master`.  It would be better, if they could reference a specific commit instead, to make sure the references don't end up pointing to the wrong code snippet as the upstream code changes.

**Comment by densmirn** [comment]
Could you please close the PR and switch to https://github.com/numba/numba/pull/4651 which includes the changes?

**Comment by esc** [comment]
@densmirn - you got it.

**Pull request closed without merging**